pkg/controllers/inventory/inventory.go

```diff
@@ -15,81 +15,62 @@
-	VpcIndexerByNamespace             = "vpc.namespace"
-	VpcIndexerByNamespacedRegion      = "namespace.region"
-
-	VpcLabelAccountName = "account-name"
-	VpcLabelRegion      = "region"
-)
-
 // InitInventory creates an instance of Inventory struct and initializes inventory with cache indexers.
 func InitInventory() *Inventory {
 	inventory := &Inventory{
 		log: logging.GetLogger("inventory").WithName("Cloud"),
 	}
-	inventory.vpcCache = cache.NewIndexer(
-		func(obj interface{}) (string, error) {
-			vpc := obj.(*runtimev1alpha1.Vpc)
-			return fmt.Sprintf("%v/%v-%v", vpc.Namespace, vpc.Labels[VpcLabelAccountName], vpc.Info.Id), nil
-		},
-		cache.Indexers{
-			VpcIndexerByAccountNameSpacedName: func(obj interface{}) ([]string, error) {
-				vpc := obj.(*runtimev1alpha1.Vpc)
-				return []string{vpc.Namespace + "/" + vpc.Labels[VpcLabelAccountName]}, nil
-			},
-			VpcIndexerByVpcNamespacedName: func(obj interface{}) ([]string, error) {
-				vpc := obj.(*runtimev1alpha1.Vpc)
-				return []string{vpc.Namespace + "/" + vpc.Name}, nil
-			},
-			VpcIndexerByNamespacedRegion: func(obj interface{}) ([]string, error) {
-				vpc := obj.(*runtimev1alpha1.Vpc)
-				return []string{vpc.Namespace + "/" + vpc.Info.Region}, nil
-			},
-			VpcIndexerByNamespace: func(obj interface{}) ([]string, error) {
-				vpc := obj.(*runtimev1alpha1.Vpc)
-				return []string{vpc.Namespace}, nil
-			},
-		})
+	inventory.vpcStore = storage.NewVPCInventoryStore()
 	return inventory
 }
 
 // BuildVpcCache using vpc list from cloud, update vpc cache(with vpcs applicable for the current cloud account).
 func (inventory *Inventory) BuildVpcCache(vpcMap map[string]*runtimev1alpha1.Vpc, namespacedName *types.NamespacedName) error {
-	vpcsInCache, _ := inventory.vpcCache.ByIndex(VpcIndexerByAccountNameSpacedName, namespacedName.String())
+	vpcsInCache, _ := inventory.vpcStore.GetByIndex(storage.VpcIndexerByAccountNameSpacedName, namespacedName.String())
 
 	// Remove vpcs in vpc cache which are not found in vpc list fetched from cloud.
 	for _, i := range vpcsInCache {
 		vpc := i.(*runtimev1alpha1.Vpc)
 		if _, found := vpcMap[vpc.Info.Id]; !found {
 			inventory.log.V(1).Info("Deleting a vpc from vpc cache", "vpc id", vpc.Info.Id, "account",
 				namespacedName.String())
-			if err := inventory.vpcCache.Delete(vpc); err != nil {
+			if err := inventory.vpcStore.Delete(fmt.Sprintf("%v/%v-%v", vpc.Namespace, vpc.Labels[storage.VpcLabelAccountName],
+				vpc.Info.Id)); err != nil {
 				inventory.log.Error(err, "failed to delete entry from VpcIndexer", "vpc id", vpc.Info.Id, "account",
 					namespacedName.String())
 			}
 		}
 	}
 
 	for _, v := range vpcMap {
-		err := inventory.vpcCache.Add(v)
+		var err error
+		key := fmt.Sprintf("%v/%v-%v", v.Namespace, v.Labels[storage.VpcLabelAccountName], v.Info.Id)
+		if _, found, _ := inventory.vpcStore.Get(key); !found {
+			err = inventory.vpcStore.Create(v)
+		} else {
+			err = inventory.vpcStore.Update(v)
```

Comment:
Does line 71 get hit(Update) in both the cases?
1. After the vpc add, when BuildVpcCache function is called again and no change in vpc object.
2. After the vpc add, when BuildVpcCache  is called and there is a change in vpc object(lets say a new tag has been added)

Reply:
sorry what is this about, I think if already in cache will go to update if not will create?